packages/calypso-e2e/src/lib/blocks/block-flows/whatsapp-button.ts

```diff
@@ -31,10 +30,14 @@ export class WhatsAppButtonFlow implements BlockFlow {
 	 */
 	constructor( configurationData: ConfigurationData ) {
 		this.configurationData = configurationData;
+		// The parent selector for the block changes between mobile and desktop viewport.
+		this.blockEditorSelector =
+			envVariables.VIEWPORT_NAME === 'desktop'
+				? 'div[aria-label="Block: WhatsApp Button"]'
+				: 'div[aria-label="Block: Send A Message"]';
```

Comment:
This solution is the best way to deal with an otherwise terrible situation.
As the comment states, the top-level selector for the WhatsApp Button block changes depending on the viewport.

Since the `EditorGutenbergComponent.addBlock` method verifies the successful insertion of a block by checking for the `${blockEditorSelector}.is-selected`, there must be a way to apply the `.is-selected` modifier to the appropriate selector based on the viewport.

This issue only became apparent during the refactor due to the use of Locator API in the refactored code.

Reply:
Nice catch!